ecommerce/models.py

```diff
@@ -81,6 +81,28 @@ def has_user_blocked_products(self, user):
             ]
         )
 
+    def has_user_purchased_same_courserun(self, user):
+        """
+            Return true if any of the courses in the basket has already purchased
+            meaning user has orders for that run in a FULFILLED or REVIEW state
+         """
+        purchased_orders = Order.objects.filter(purchaser=user).filter(state__in=[Order.STATE.FULFILLED, Order.STATE.REVIEW])
```

Comment:
This works but I think there's a better way to do this. If the user's successfully paid for and enrolled in the course, there should be a CourseRunEnrollment that is in the `verified` state. Can this check for that instead? This would get a few cases where checking for a completed order might be a bit problematic. (Thinking specifically of the case where a learner may have been refunded for an order for a course run but we're not changing their enrollment status - in that case, they probably shouldn't be able to pay for the course run again.)

Reply:
@jkachel Thanks for the feedback. Do you mean that it should check for CourseRunEnrollment.enrollment_mode = 'verified' instead of ecommerce_order.status in ('fulfilled', 'reviewed')? That works for my test enrollment on RC (I can see mine has enrollment_mode = 'verified'), but when I checked the 3 orders peter linked in the ticket, they all have 'audit' state, not verified. Is this just data issues for these 3 orders?
![image](https://user-images.githubusercontent.com/3138890/180828665-c6f46f54-90ec-4077-aafd-746aa5418098.png)
I can update the logic, just wanted to confirm.

Reply:
Like Rachel mentioned, we have some situations that have come up where someone has paid but isn't enrolled as verified. That's what we're trying to avoid here so that if this happens a learner doesn't pay repeatedly and will instead contact support.